go/vt/vtgate/planbuilder/operators/update.go

```diff
@@ -107,24 +116,70 @@ func createOperatorFromUpdate(ctx *plancontext.PlanningContext, updStmt *sqlpars
 		Lock:     sqlparser.ShareModeLock,
 	}
 
-	parentFks := ctx.SemTable.GetParentForeignKeysList()
-	childFks := ctx.SemTable.GetChildForeignKeysList()
 	if len(childFks) == 0 && len(parentFks) == 0 {
 		return op
 	}
 
-	// If the delete statement has a limit, we don't support it yet.
-	if updStmt.Limit != nil {
-		panic(vterrors.VT12001("update with limit with foreign key constraints"))
+	return buildFkOperator(ctx, op, updClone, parentFks, childFks, vTbl)
+}
+
+func updateWithInputPlanningForFk(ctx *plancontext.PlanningContext, upd *sqlparser.Update) Operator {
+	updClone := ctx.SemTable.Clone(upd).(*sqlparser.Update)
+	upd.Limit = nil
+
+	selectStmt := &sqlparser.Select{
+		From:    updClone.TableExprs,
```

Comment:
`upd.OrderBy = nil` can also be added here

Reply:
That causes end to end tests to fail, because ordering of updates is important when the updates are cascaded.